### DISCUSSION TITLE
method.getArguments().get(1)) for String

### DISCUSSION THREAD

**Discussion by Lipuxy** [openrewrite/rewrite#3803] [Q&A]
https://github.com/moderneinc/rewrite-recipe-starter/blob/main/src/main/java/com/yourorg/NoGuavaListsNewArrayList.java 
Line 85
method.getArguments().get(0));
The above line works fine for int 
but if we pass String argument for method signature methodname(int,String,String,String)
and try to access argument 2 by using method.getArguments().get(1));
it fails .
 What should be the alternative code to access String here ?

**Comment by timtebeek** [comment]
Hi @Lipuxy ; it's not immediately clear to me what you're trying to achieve; are you developing your own recipe based on that starter?

The recipe in the starter
https://github.com/moderneinc/rewrite-recipe-starter/blob/main/src/main/java/com/yourorg/NoGuavaListsNewArrayList.java
is a copy of the recipe seen here
https://github.com/openrewrite/rewrite-migrate-java/blob/main/src/main/java/org/openrewrite/java/migrate/guava/NoGuavaListsNewArrayList.java

If you mean to say there's a bug in the recipe implementation the best way to report that is over on rewrite-migrate-java:
https://github.com/openrewrite/rewrite-migrate-java/issues/new/choose

If you're looking to develop your own recipe it would help to know what changes you're looking to make, and what does and does not work. A fork or test case would then be most helpful, as from the details shared above I don't know what you've already changed, or are looking to do.

**Reply by Lipuxy** [reply]
yes..you are right ..I am trying to develop my own recipe using teh below starter 
https://github.com/moderneinc/rewrite-recipe-starter/blob/main/src/main/java/com/yourorg/NoGuavaListsNewArrayList.java 

My requirement is to change 
Class1.method1(int, String2, String, String4)
to
throw new Class2(String 2, String4)

By using the above github reference , if I want to access String2 and String4 in the below format for eg
return newArrayListCapacity.apply(getCursor(), method.getCoordinates().replace(),
                            method.getArguments().get(1), method.getArguments().get(3));

It does not work.

But in teh github it is able to fetch the 1st element by using 
return newArrayListCapacity.apply(getCursor(), method.getCoordinates().replace(),
                            method.getArguments().get(0));

**Reply by timtebeek** [reply]
That sounds like you're pretty much writing a whole new recipe, given how much I suspect needs to change. Did you update your MethodMatcher fields and Preconditions already to match the new method you're matching?

- This page might help you to match method patterns: https://docs.openrewrite.org/reference/method-patterns

- And then these pages help you to write recipes: https://docs.openrewrite.org/authoring-recipes

- In particular I recommend you start with writing a test: https://docs.openrewrite.org/authoring-recipes/recipe-testing
That way a number of common mistakes are checked by our testing support framework, and you can step through with the debugger.

I hope that helps! If you have more questions I recommend you to start a reproducer project, and share that here or in our public Slack.

**Reply by Lipuxy** [reply]
For the below requirement:
 change
Class1.method1(int, String2, String3, String4)
to
throw new UseDefinedException(String2)

In my recipe I have added the return as 
return myTemplate.apply(getCursor(), method.getCoordinates().replace(),
method.getArguments().get(1));
When I write junit test cases to test the recipe I am getting the exception  : java.lang.IllegalArgumentException : Failed to parse as Java..what could be the issue ?

**Reply by timtebeek** [reply]
Hard to say without seeing how you build up your JavaTemplate String, JavaParser classpath and imports. Maybe best to join [our Slack](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA) for some synchronous help?

**Comment by timtebeek** [comment]
Closing as stale; feel free to open a new discussion if this is still relevant.
